### ISSUE TITLE
Non-conforming data output

### ISSUE THREAD

**Issue by mylesj** [json-schema-faker/json-schema-faker#740]
Hi, I've been trying to narrow down some unexpected output and wondered if you could offer any insight into why I'm seeing this - I'm reasonably sure that the schema is correct, it was generated by `ts-json-schema-generator` for the following typescript type: 

```typescript
type Dict<T> = { [key: string]: T }

type Item = {
    value: string
}

export type Test = {
    items: Item[],
    itemsById: Dict<Item>
}
```

<details>
<summary>schema.json</summary>

```json
{
    "$schema": "http://json-schema.org/draft-07/schema#",
    "$ref": "#/definitions/Test",
    "definitions": {
        "Test": {
            "type": "object",
            "properties": {
                "items": {
                    "type": "array",
                    "items": {
                        "type": "object",
                        "properties": {
                            "value": {
                                "type": "string"
                            }
                        },
                        "required": [
                            "value"
                        ],
                        "additionalProperties": false
                    }
                },
                "itemsById": {
                    "type": "object",
                    "additionalProperties": {
                        "type": "object",
                        "properties": {
                            "value": {
                                "type": "string"
                            }
                        },
                        "required": [
                            "value"
                        ],
                        "additionalProperties": false
                    }
                }
            },
            "required": [
                "items",
                "itemsById"
            ],
            "additionalProperties": false
        }
    }
}
```
</details>

<details>
<summary>$ node index.js</summary>

```javascript
const { readFile } = require('fs/promises')
const { JSONSchemaFaker: jsf } = require('json-schema-faker')

jsf.option({
    alwaysFakeOptionals: true,
    fillProperties: true,
})

main();
async function main() {
    const schema = JSON.parse(await readFile('./schema.json'))
    const data = await jsf.resolve(schema)
    console.log(JSON.stringify(data, null, 2))
}
```
</details>

When I try to generate data locally, I see primitive values against keys in the `itemsById` field: 

```json
{
  "items": [
    // this is correct 
    {
      "value": "incididunt cillum labore est amet"
    }
  ],
  "itemsById": {
    "nisi_1c": 84280283.71798155,  // incorrect (number)  
    "sunt4b": false,               // incorrect (boolean) 
    "est7": {                      // correct             
      "value": "ipsum"
    }
  }
}
```

Where I expect to see a consistent `Item` shape: 

```json
{
  "items": [
    // ... 
  ],
  "itemsById": {
    "<rand: string>": {            
      "value": "<rand: string>"
    },
    "<rand: string>": {            
      "value": "<rand: string>"
    }
  }
}
```

It confused me that the webpage https://json-schema-faker.js.org/ seems to generate the data I expected consistently - is this an issue with the module import or am I missing configuration? I also did checkout the `develop` branch but I'm seeing the same issue there too.

**Comment by pateketrueke** [comment]
Hi, the webapp has the `0.5.0-rcv.45` version, which version is installed in your nodejs?

**Comment by mylesj** [comment]
I was test RC `46` - but I see the same issue if I rollback a version too.

**Comment by pateketrueke** [comment]
Yeah, definitely this is a regression.

Gonna investigate this asap, thank you!

**Comment by liangskyli** [comment]
this bug reproduce: webpage https://json-schema-faker.js.org/ , set alwaysFakeOptions true to reproduce
I have fix: #757 
the same bug: #749 
@pateketrueke This bug can be closed now！
